src/sentry/models/organizationmember.py

```diff
@@ -102,12 +109,31 @@ class Meta:
     def save(self, *args, **kwargs):
         assert self.user_id or self.email, \
             'Must set user or email'
+        if self.token:
+            now = timezone.now()
+            self.token_expires_at = now + timedelta(days=INVITE_DAYS_VALID)
         super(OrganizationMember, self).save(*args, **kwargs)
 
+    def set_user(self, user):
+        self.user = user
+        self.email = None
+        self.token = None
+        self.token_expires_at = None
+
     @property
     def is_pending(self):
         return self.user_id is None
 
+    @property
+    def token_expired(self):
+        # Old tokens don't expire to preserve compatibility and not require
+        # a backfill migration.
+        if self.token_expires_at is None:
```

Comment:
we'd probably want to change this in the future (e.g. after a week or so auto expire tokens)

alternatively we could just do the backfill easy enough

Reply:
also one thing to note, in LostPasswordHash we do the inverse pattern -- we track date_created and make it so you can dynamically change "how long are these valid" since its just compute-time-since-start vs compute-end-date-on-creation

Reply:
We have `date_added` on the organizationmember, but that value doesn't update when an invitation is re-sent. I would have needed a second column to track the last send date instead. I'm happy to change that around if you'd prefer this follows the pattern set elsewhere.